openml/tasks/functions.py

```diff
@@ -23,21 +23,19 @@ def _get_cached_tasks():
         # description
 
         for filename in directory_content:
-            match = re.match(r"(tid)_([0-9]*)\.xml", filename)
-            if match:
-                tid = match.group(2)
-                tid = int(tid)
+            if not re.match(r"[0-9]*", filename):
+                continue
 
-                tasks[tid] = _get_cached_task(tid)
+            tid = int(filename)
+            tasks[tid] = _get_cached_task(tid)
 
     return tasks
 
 
 def _get_cached_task(tid):
     for cache_dir in [config.get_cache_directory(), config.get_private_directory()]:
         task_cache_dir = os.path.join(cache_dir, "tasks")
-        task_file = os.path.join(task_cache_dir,
-                                 "tid_%d.xml" % int(tid))
+        task_file = os.path.join(task_cache_dir, str(tid), "task.xml")
```

Comment:
so you changed the file names? Or was it broken before? This breaks all the users caches (if anyone uses this) ;)